src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,60 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import {
+  getLinkDataCy,
+  getLinkHref,
+  getLinkTextContent,
+  handleClickPagination,
+} from './pagination-utils';
+
+export interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
```

Comment:
Ensure that the calculation of `pagesAmount` handles cases where `perPage` is 0 to prevent division by zero errors. You may want to add a guard clause or a default fallback value.